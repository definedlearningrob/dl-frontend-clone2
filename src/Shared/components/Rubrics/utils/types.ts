export type RubricData = {
  criteriaLabels: RubricCriteriaLabel[];
  criterias: RubricCriteria[];
  description?: string;
  headings: RubricHeading[];
  id: string;
  name: string;
  pointsAvailable?: number;
};

export type RubricCriteriaLabel = {
  displayName: string | null;
  id: string;
  score: number;
};

export type RubricCriteria = {
  id?: string;
  rubricCriteriaLabelId: string;
  rubricHeadingId: string;
  text: string;
  uuid?: string;
};

export type RubricHeading = {
  id: string;
  multiplier: number;
  name: string;
};

export type FillMissingCriteriasArguments = {
  criteriaLabels: RubricCriteriaLabel[];
  criterias: RubricCriteria[];
  headings: RubricHeading[];
};

export type RubricHeadingIdPairs = {
  headingId: string;
  criteriaLabelId: string;
}[];
