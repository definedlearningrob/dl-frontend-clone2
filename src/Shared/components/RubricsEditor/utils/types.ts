import { TTag } from '@dc/graphql/user/queries/tag';
import { TRubricHeadingPlan } from '@dc/graphql/user/queries/rubric';

export type RubricData = {
  criteriaLabels: RubricCriteriaLabel[];
  criterias: RubricCriteria[];
  description?: string;
  headings: RubricHeading[];
  id: string;
  name: string;
};

export type RubricEdit = RubricData & { canEdit: boolean };

export type RubricCriteriaLabel = {
  displayName: string | null;
  id: string;
  score: number;
};

export type RubricCriteria = {
  id?: string;
  rubricCriteriaLabelId: string;
  rubricHeadingId: string;
  text: string;
  uuid?: string;
};

export type RubricHeading = {
  id: string;
  multiplier: number;
  name: string;
  tags?: TTag[];
  plans?: TRubricHeadingPlan[];
  statements: { id: string }[];
};

export type FillMissingCriteriasArguments = {
  criteriaLabels: RubricCriteriaLabel[];
  criterias: RubricCriteria[];
  headings: RubricHeading[];
};

export type RubricHeadingIdPairs = {
  headingId: string;
  criteriaLabelId: string;
}[];
