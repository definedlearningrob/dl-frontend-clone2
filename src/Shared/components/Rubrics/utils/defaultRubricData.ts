export const defaultRubricData = {
  headings: [
    { id: 'heading-1', name: 'rubric.careerPathway.name', multiplier: 1 },
    { id: 'heading-2', name: 'rubric.projectGoal.name', multiplier: 1 },
    { id: 'heading-3', name: 'rubric.projectRole.name', multiplier: 1 },
    { id: 'heading-4', name: 'rubric.jobProductQuality.name', multiplier: 1 },
  ],
  criteriaLabels: [
    { displayName: 'test label', id: 'criteria-label-1', score: 1 },
    { displayName: null, id: 'criteria-label-2', score: 2 },
    { displayName: null, id: 'criteria-label-3', score: 3 },
    { displayName: null, id: 'criteria-label-4', score: 4 },
  ],
  criterias: [
    {
      id: 'criteria-1-1',
      rubricCriteriaLabelId: 'criteria-label-1',
      rubricHeadingId: 'heading-1',
      text: 'rubric.careerPathway.1',
    },
    {
      id: 'criteria-1-2',
      rubricCriteriaLabelId: 'criteria-label-2',
      rubricHeadingId: 'heading-1',
      text: 'rubric.careerPathway.2',
    },
    {
      id: 'criteria-1-3',
      rubricCriteriaLabelId: 'criteria-label-3',
      rubricHeadingId: 'heading-1',
      text: 'rubric.careerPathway.3',
    },
    {
      id: 'criteria-1-4',
      rubricCriteriaLabelId: 'criteria-label-4',
      rubricHeadingId: 'heading-1',
      text: 'rubric.careerPathway.4',
    },
    {
      id: 'criteria-2-1',
      rubricCriteriaLabelId: 'criteria-label-1',
      rubricHeadingId: 'heading-2',
      text: 'rubric.projectGoal.1',
    },
    {
      id: 'criteria-2-2',
      rubricCriteriaLabelId: 'criteria-label-2',
      rubricHeadingId: 'heading-2',
      text: 'rubric.projectGoal.2',
    },
    {
      id: 'criteria-2-3',
      rubricCriteriaLabelId: 'criteria-label-3',
      rubricHeadingId: 'heading-2',
      text: 'rubric.projectGoal.3',
    },
    {
      id: 'criteria-2-4',
      rubricCriteriaLabelId: 'criteria-label-4',
      rubricHeadingId: 'heading-2',
      text: 'rubric.projectGoal.4',
    },
    {
      id: 'criteria-3-1',
      rubricCriteriaLabelId: 'criteria-label-1',
      rubricHeadingId: 'heading-3',
      text: 'rubric.projectRole.1',
    },
    {
      id: 'criteria-3-2',
      rubricCriteriaLabelId: 'criteria-label-2',
      rubricHeadingId: 'heading-3',
      text: 'rubric.projectRole.2',
    },
    {
      id: 'criteria-3-3',
      rubricCriteriaLabelId: 'criteria-label-3',
      rubricHeadingId: 'heading-3',
      text: 'rubric.projectRole.3',
    },
    {
      id: 'criteria-3-4',
      rubricCriteriaLabelId: 'criteria-label-4',
      rubricHeadingId: 'heading-3',
      text: 'rubric.projectRole.4',
    },
    {
      id: 'criteria-4-1',
      rubricCriteriaLabelId: 'criteria-label-1',
      rubricHeadingId: 'heading-4',
      text: 'rubric.jobProductQuality.1',
    },
    {
      id: 'criteria-4-2',
      rubricCriteriaLabelId: 'criteria-label-2',
      rubricHeadingId: 'heading-4',
      text: 'rubric.jobProductQuality.2',
    },
    {
      id: 'criteria-4-3',
      rubricCriteriaLabelId: 'criteria-label-3',
      rubricHeadingId: 'heading-4',
      text: 'rubric.jobProductQuality.3',
    },
    {
      id: 'criteria-4-4',
      rubricCriteriaLabelId: 'criteria-label-4',
      rubricHeadingId: 'heading-4',
      text: 'rubric.jobProductQuality.4',
    },
  ],
};
