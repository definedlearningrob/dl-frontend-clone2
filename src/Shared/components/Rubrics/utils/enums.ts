export enum RUBRIC_TYPE {
  VIEWER = 'VIEWER',
  GRADER = 'GRADER',
  PREVIEW = 'PREVIEW',
}
