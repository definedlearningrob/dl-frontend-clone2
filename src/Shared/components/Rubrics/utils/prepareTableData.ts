import { v4 as uuid } from 'uuid';

import {
  FillMissingCriteriasArguments,
  RubricHeadingIdPairs,
  RubricCriteria,
  RubricData,
} from './types';

const fillMissingCriteria = ({
  headings,
  criteriaLabels,
  criterias,
}: FillMissingCriteriasArguments): RubricCriteria[] => {
  const labelIds = criteriaLabels.map((label) => label.id);
  const headingIds = headings.map((heading) => heading.id);

  const labelHeadingIdPairs: RubricHeadingIdPairs = labelIds.reduce(
    (acc: RubricHeadingIdPairs, criteriaLabelId) => {
      const pairsForFirst = headingIds.map((headingId) => ({ headingId, criteriaLabelId }));
      acc.push(...pairsForFirst);

      return acc;
    },
    []
  );

  return labelHeadingIdPairs.map((pair) => {
    const foundCriteria = criterias.find(
      ({ rubricCriteriaLabelId, rubricHeadingId }) =>
        rubricCriteriaLabelId === pair.criteriaLabelId && rubricHeadingId === pair.headingId
    );

    return (
      foundCriteria || {
        uuid: uuid(),
        text: '',
        rubricCriteriaLabelId: pair.criteriaLabelId,
        rubricHeadingId: pair.headingId,
      }
    );
  });
};

export const prepareTableData = (rubric: RubricData): RubricData => {
  const { criterias, headings, criteriaLabels } = rubric;

  return {
    ...rubric,
    criterias: fillMissingCriteria({ headings, criteriaLabels, criterias }),
  };
};
