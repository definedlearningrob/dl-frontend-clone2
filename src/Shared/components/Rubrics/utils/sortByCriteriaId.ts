import { RubricCriteria, RubricCriteriaLabel } from './types';

export const byCriteriaLabelIds =
  (criteriaLabels: RubricCriteriaLabel[]) =>
  (criteriaA: RubricCriteria, criteriaB: RubricCriteria) =>
    criteriaLabels.findIndex((label) => label.id === criteriaA.rubricCriteriaLabelId) -
    criteriaLabels.findIndex((label) => label.id === criteriaB.rubricCriteriaLabelId);
