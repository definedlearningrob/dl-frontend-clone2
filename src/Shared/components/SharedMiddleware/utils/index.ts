export * from './parseSharedResource';
export * from './getResourceLink';
