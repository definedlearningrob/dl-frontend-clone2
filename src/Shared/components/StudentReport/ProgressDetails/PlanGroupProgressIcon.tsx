import { IconContainer } from '@shared/components/IconContainer/IconContainer';
import { cx } from '@shared/utils/cx';
import { TPlanStatusOption } from '@shared/hooks/usePlanStatusOptions';

type Props = {
  item: TPlanStatusOption;
};

export const PlanGroupProgressIcon = ({ item }: Props) => (
  <IconContainer
    Icon={item.Icon}
    className={cx('bg-white outline outline-1 outline-neutral-300 rounded-xs mr-xxs', {
      'text-neutral-700': item.variant === 'neutral',
      'text-secondary-500': item.variant === 'secondary',
      'text-success-500': item.variant === 'success',
      'text-danger-500': item.variant === 'danger',
    })}
    paddingSize='xxs'
    size='sm'
  />
);
