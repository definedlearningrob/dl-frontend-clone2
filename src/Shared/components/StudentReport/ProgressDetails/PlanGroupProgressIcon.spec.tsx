import { renderWithI18N } from '@shared/utils/renderWithI18N';
import { TPlanStatusOption } from '@shared/hooks/usePlanStatusOptions';

import { PlanGroupProgressIcon } from './PlanGroupProgressIcon';

const variants = [
  { variant: 'neutral', label: 'Not started', value: 'not_started', testId: 'neutral-icon' },
  { variant: 'secondary', label: 'In progress', value: 'in_progress', testId: 'secondary-icon' },
  { variant: 'success', label: 'Completed', value: 'completed', testId: 'success-icon' },
  { variant: 'danger', label: 'Not met', value: 'not_met', testId: 'danger-icon' },
];

describe('PlanGroupProgressIcon component', () => {
  variants.forEach(({ variant, label, value, testId }) => {
    it(`renders with ${variant} variant correctly`, () => {
      const item = {
        Icon: () => <svg data-testid={testId} />,
        variant: variant as TPlanStatusOption['variant'],
        label,
        value,
      };

      const { container } = renderWithI18N(<PlanGroupProgressIcon item={item} />, 'DC');

      expect(container).toMatchSnapshot();
    });
  });
});
