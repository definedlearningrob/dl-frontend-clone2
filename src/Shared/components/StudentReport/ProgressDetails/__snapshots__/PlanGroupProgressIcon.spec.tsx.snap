// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanGroupProgressIcon component renders with danger variant correctly 1`] = `
<div>
  <div
    class="flex"
  >
    <div
      class="flex align-center justify-center grow-0 shrink-0 bg-white outline outline-1 outline-neutral-300 rounded-xs mr-xxs text-danger-500 p-xxs"
      data-testid="icon"
    >
      <svg
        data-testid="danger-icon"
      />
    </div>
  </div>
</div>
`;

exports[`PlanGroupProgressIcon component renders with neutral variant correctly 1`] = `
<div>
  <div
    class="flex"
  >
    <div
      class="flex align-center justify-center grow-0 shrink-0 bg-white outline outline-1 outline-neutral-300 rounded-xs mr-xxs text-neutral-700 p-xxs"
      data-testid="icon"
    >
      <svg
        data-testid="neutral-icon"
      />
    </div>
  </div>
</div>
`;

exports[`PlanGroupProgressIcon component renders with secondary variant correctly 1`] = `
<div>
  <div
    class="flex"
  >
    <div
      class="flex align-center justify-center grow-0 shrink-0 bg-white outline outline-1 outline-neutral-300 rounded-xs mr-xxs text-secondary-500 p-xxs"
      data-testid="icon"
    >
      <svg
        data-testid="secondary-icon"
      />
    </div>
  </div>
</div>
`;

exports[`PlanGroupProgressIcon component renders with success variant correctly 1`] = `
<div>
  <div
    class="flex"
  >
    <div
      class="flex align-center justify-center grow-0 shrink-0 bg-white outline outline-1 outline-neutral-300 rounded-xs mr-xxs text-success-500 p-xxs"
      data-testid="icon"
    >
      <svg
        data-testid="success-icon"
      />
    </div>
  </div>
</div>
`;
